torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -553,23 +553,25 @@ Tensor unsqueeze_to(const Tensor & self, int64_t dim, IntArrayRef sizes) {
   return self;
 }
 
-std::vector<Tensor> cat_tensors_backward(const Tensor & grad, const std::vector<std::vector<int64_t>> &sizes, int64_t dim) {
+std::vector<Tensor> cat_tensors_backward(const Tensor & grad, const std::vector<std::vector<int64_t>> &sizes, const std::vector<ScalarType> &dtypes, int64_t dim) {
   std::vector<Tensor> grad_inputs(sizes.size());
   if (!grad.defined()) {
     return grad_inputs;
   }
   dim = at::legacy_cat_wrap_dim(dim, sizes);
   int64_t accumulate = 0;
+
   for (size_t i = 0; i < sizes.size(); ++i) {
     auto& shape = sizes[i];
+    auto grad_ = handle_r_to_c(dtypes[i], grad);
```

Comment:
I guess you don't really need to do this each time through (i.e. you only need at most 1 real view of the grad).  But probably doesn't matter performance wise.